File: Hackerrank/Code Heat/suits.py
from collections import Counter as Count

for k in range(int(raw_input())):
    n = int(raw_input())
    nos = map(int, raw_input().split())

    temp = [0]

    for i in range(len(nos)):
        if i == 0:
            temp.append(nos[i])
        else:
            temp.append(nos[i] + temp[i])

    d = Count(temp)
    print d

File: Hackerrank/Code Heat/suits.cpp
#include <iostream>

using namespace std;

int main()
{
    int t;
    cin>>t;

    while(t--)
    {
        int n;
        cin>>n;

        long long int no[n], temp[n];
        temp[0] = 0;

        for(int i = 0; i < n; i++)
        {
            cin>>no[i];

            if(i == 0)
                temp[i + 1] = no[i];
            else
                temp[i + 1] = no[i] + temp[i];
        }



    }
    return 0;
}